File: php/postTarget.php
<?php

/**
 * Ce script permet de créer une cible.
 *
 * @package cookbook
 */

require 'utils.php';

/**
 * Variable contenant les configurations pour se connecter à l'API
 *
 * @var array
 */
$configs = parse_ini_file("config.ini");
/**
 * Creation du JSON en fonction des champs que vous voulez renseigner.
 * Remplacez les XXXX par l'id des champs que vous voulez renseigner.
 * Les valeurs renseignez ici sont à remplacer par les votres.
 *
 * @var array
 */
$data = array(
							"XXXX" => "Toto",
							"XXXX" => "Mr",
							"XXXX" => "test@test.com",
							"XXXX" => "0123456789",
							"XXXX" => "150 caracters max",
							"XXXX" => 123,
							"XXXX" => "01/01/2000",
							"XXXX" => array("valeur 1", "valeur 2"),
);
$dataJson = json_encode($data);


/**
 * On créé le liens pour verifier si la cible n'existe pas, remplacez XXXX par
 * l'id d'un champ d'unicité.
 */
$url = $configs['url'] . 'targets?unicity=' . $data['XXXX'];

$con = connect($url, $configs['xKey'], null, 'GET');

// Verification des reponses
if ($con['result'] == false) {
  // La cible n'éxiste pas, nous devons la créer

  // Affichage de l'erreur
  echo "Error : " .
       $con['info']['http_code'] .
       " : Creation of the target.\n";

  $con = connect($url, $configs['xKey'], $dataJson, 'POST');
	echo "The target " .
			 $data['XXXX'] .
			 " has been created.";
}
else {
  // La cible existe, nous la mettons a jour

  // On affiche les anciennes valeurs
  echo $con['result'] . "\n";

  // On remplit la requete 'PUT'
  $con = connect($url, $configs['xKey'], $dataJson, 'PUT');
  echo "The target " .
       $data['XXXX'] .
       " has been updated.";
}

?>

File: Csharp/postTarget.cs
using Newtonsoft.Json.Linq;
using System;
using System.Net;
using System.Xml;


namespace postTarget
{
    internal class Program
    {
        private static void Main(string[] args)
        {
            // Changez le Path pour correspondre à la destination de votre fichier de configuration
            var doc = new XmlDocument();
            doc.Load("config.xml");

            var xKey = doc.DocumentElement.SelectSingleNode("/config/xKey").InnerText;
            var baseUrl = doc.DocumentElement.SelectSingleNode("/config/url").InnerText;

            //Syntaxe pour les differents types d'informations :
            var chain = "name";	//Chaine de caracteres
            var listOfValues = "Mr";	//Liste de valeurs
            var email = "test@test.com";	//E-mail
            var phoneNumber = "0123456789";	//Telephone
            var textZone = "150 caracters max";	//Zone de texte
            var numbers = 123;	//Valeur numerique
            var date = "01/01/2000";	//Date
            var listMultipleValues = new JProperty("XXXX", "valeur 1", "valeur 2");	//Liste de valeurs multiples : particulier, la premiere valeur est l'id-champ, les autres sont les valeurs de ce champ.

            //Creation du tableau en fonction de l'id des champs de la fiche cible : "id-champ" => "valeur du champ"
            JObject data = new JObject();
            data.Add("XXXX", chain);
            data.Add("XXXX", listOfValues);
            data.Add("XXXX", email);
            data.Add("XXXX", phoneNumber);
            data.Add("XXXX", textZone);
            data.Add("XXXX", numbers);
            data.Add("XXXX", date);
            data.Add(listMultipleValues);   //Liste de valeurs multiples : particulier, la premiere valeur est l'id-champ, les autres sont les valeurs de ce champ.


            //Lancement de la connexion pour remplir la requete 'GET'
            var url = baseUrl + "targets?unicity=" + email;
            //Requette 'POST' pour creer la cible
            var reponse = Utils.allConnection(url, xKey, data, "POST");

            //Verification des reponses
            if ((int)reponse[0] == 200)
            {
                Console.Write("{0}", (string)reponse[3]);
            }
            else
            {
                //Affichage de l'erreur
                Console.Write("Error : {0} {1}", (int)reponse[0], ((HttpWebResponse)reponse[2]).StatusCode.ToString());
            }

            //Attente de lecture (optionnel)
            Console.ReadLine();

            ((HttpWebResponse)reponse[2]).Close();
        }

    }
}
